src/sinks/util/buffer/vec.rs

```diff
@@ -0,0 +1,49 @@
+use super::super::{Batch, BatchSettings, PushResult};
+
+#[derive(Clone)]
+pub struct VecBuffer<T> {
+    batch: Vec<T>,
+    max_events: usize,
+}
+
+impl<T> VecBuffer<T> {
+    pub fn new(settings: BatchSettings) -> Self {
+        Self {
+            batch: Vec::with_capacity(settings.events),
+            max_events: settings.events,
+        }
+    }
+}
+
+impl<T> Batch for VecBuffer<T> {
+    type Input = T;
+    type Output = Vec<T>;
+
+    fn push(&mut self, item: Self::Input) -> PushResult<Self::Input> {
+        if self.batch.len() >= self.max_events {
+            PushResult::Full(item)
+        } else {
+            self.batch.push(item);
+            PushResult::Ok
+        }
+    }
+
+    fn is_empty(&self) -> bool {
+        self.batch.is_empty()
+    }
+
+    fn fresh(&self) -> Self {
+        Self {
+            batch: Vec::with_capacity(self.max_events),
+            max_events: self.max_events,
+        }
+    }
```

Comment:
There few places where code in `new` and `fresh` is same. Maybe it worth add non-pub `new2` function?

For example for `VecBuffer` it will looks like:
```rust
impl<T> VecBuffer<T> {
    pub fn new(settings: BatchSettings) -> Self {
        VecBuffer::new2(settings.events)
    }
    fn new2(max_events: usize) -> Self {
        let batch = Vec::with_capacity(max_events);
        Self { batch, max_events, }
    }
}
impl<T> Batch for VecBuffer<T> {
    fn fresh(&self) -> Self {
        VecBuffer::new2(self.max_events)
    }
}
```

Reply:
Yes, that's worth considering.